### DISCUSSION TITLE
Getting an association ID for c_store_scp for differentiating

### DISCUSSION THREAD

**Discussion by arnond** [fo-dicom/fo-dicom#1533] [Q&A]
Hi All,

I am writing a storage SCP and I like to save the incoming objects per association in a different folder. How do I differentiate?

public async Task<DicomCStoreResponse> OnCStoreRequestAsync(DicomCStoreRequest request)
        {
          //I need somwthing here, I do not want to use patientID or studyInstanceUID.
            return new DicomCStoreResponse(request, DicomStatus.Success);
        }

**Comment by gofal** [comment]
Your class, where you implement the OnCStoreRequestAsync, inherits from `DicomService`. When an Associtation is opened, then the DicomServer will create a new instance of your class, and every Callback will be injected into that instance per Association. 
This means: you can add some private fields to your class. And in method `OnReceiveAssociationRequestAsync`, where you accept the incomming association, you can set some unique value to  this fields, like the timestamp, or a guid, or an increasing number. And then within the method `OnCStoreRequestAsync` you can access this field to get a unique folder per association.

By default, the `DicomService` class has a property `Association` that contains information like the sending AET, remote IP / port etc. so if this is enough information to create different folders, then you can also use that.
